android/src/main/java/com/tailscale/ipn/ui/viewModel/IpnViewModel.kt

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package com.tailscale.ipn.ui.viewModel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.tailscale.ipn.mdm.MDMSettings
+import com.tailscale.ipn.ui.localapi.Client
+import com.tailscale.ipn.ui.model.Ipn
+import com.tailscale.ipn.ui.model.IpnLocal
+import com.tailscale.ipn.ui.notifier.Notifier
+import com.tailscale.ipn.ui.util.set
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * Base model for most models in this application. Provides common facilities for watching IPN
+ * notifications, managing login/logout, updating preferences, etc.
+ */
+open class IpnViewModel : ViewModel() {
+    companion object {
+        val mdmSettings: StateFlow<MDMSettings> = MutableStateFlow(MDMSettings())
```

Comment:
mdmSettings are now modeled as a StateFlow for consistency with how we manage other state.